packages/office-ui-fabric-react/src/components/KeytipLayer/KeytipLayer.base.tsx

```diff
@@ -0,0 +1,205 @@
+  public static defaultProps: IKeytipLayerProps = {
+    keytipStartSequences: [defaultStartSequence],
+    keytipExitSequences: [defaultExitSequence],
+    keytipReturnSequences: [defaultReturnSequence],
+    content: 'Alt Windows'
+  };
+
+  private _keytipManager: KeytipManager = KeytipManager.getInstance();
+  private _classNames: {[key in keyof IKeytipLayerStyles]: string };
+
+  // tslint:disable-next-line:no-any
+  constructor(props: IKeytipLayerProps, context: any) {
+    super(props, context);
+
+    this.state = {
+      inKeytipMode: false,
+      // Get the initial set of keytips
+      keytips: [...this._keytipManager.getKeytips()],
+    };
+
+    this._keytipManager.init(this);
+  }
+
+  /**
+   * Sets the keytips state property
+   *
+   * @param keytipProps - Keytips to set in this layer
+   */
+  public setKeytips(keytipProps: IKeytipProps[]) {
+    this.setState({ keytips: keytipProps });
+  }
+
+  public render(): JSX.Element {
+    const {
+      content,
+      getStyles
+    } = this.props;
+
+    const {
+      keytips
+    } = this.state;
+
+    this._classNames = getClassNames(
+      getStyles!
+    );
+
+    return (
+      <Layer getStyles={ getLayerStyles }>
+        <span id={ ktpLayerId } className={ this._classNames.innerContent }>{ content }</span>
+        <span id={ ktpAriaSeparatorId } className={ this._classNames.innerContent }>{ ktpAriaSeparator }</span>
+        { keytips && keytips.map((keytipProps: IKeytipProps, index: number) => {
+          return <Keytip key={ index } { ...keytipProps } />;
+        }) }
+      </Layer>
+    );
+  }
+
+  public componentDidMount(): void {
+    // Add window listeners
+    this._events.on(window, 'mousedown', this._onDismiss);
+    this._events.on(window, 'resize', this._onDismiss);
+    this._events.on(window, 'keydown', this._onKeyDown, true /* useCapture */);
+    this._events.on(window, 'keypress', this._onKeyPress, true /* useCapture */);
+    this._events.on(window, 'scroll', this._onDismiss, true /* useCapture */);
+  }
+
+  public componentWillUnmount(): void {
+    // Remove window listeners
+    this._events.off(window, 'mousedown', this._onDismiss);
+    this._events.off(window, 'resize', this._onDismiss);
+    this._events.off(window, 'keydown', this._onKeyDown, true /* useCapture */);
+    this._events.off(window, 'keypress', this._onKeyPress, true /* useCapture */);
+    this._events.off(window, 'scroll', this._onDismiss, true /* useCapture */);
+  }
+
+  /**
+   * Exits keytip mode for this layer
+   */
+  public exitKeytipMode(): void {
+    if (this.props.onExitKeytipMode) {
+      this.props.onExitKeytipMode();
+    }
+    this.setState({ inKeytipMode: false });
+  }
+
+  /**
+   * Enters keytip mode for this layer
+   */
+  public enterKeytipMode(): void {
+    if (this.props.onEnterKeytipMode) {
+      this.props.onEnterKeytipMode();
+    }
+    this.setState({ inKeytipMode: true });
+  }
+
+  @autobind
```

Comment:
>@autobind [](start = 2, length = 9)

I learned on another CR that fabric is moving away from using this, so FYI, searching around the code I think the syntax used here is something like:
private _onDismiss = (ev: React.MouseEvent<HTMLElement>): void  => {
....

but dont quote me as an expert here :)

Reply:
Yeah I think I remember seeing that too...I will follow up